app/src/main/java/com/humberto/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -6,36 +6,65 @@ import com.humberto.tasky.agenda.domain.event.Attendee
 import com.humberto.tasky.agenda.domain.event.EventRepository
 import com.humberto.tasky.core.data.networking.safeCall
 import com.humberto.tasky.core.database.dao.EventDao
+import com.humberto.tasky.core.domain.repository.SessionManager
 import com.humberto.tasky.core.domain.util.DataError
 import com.humberto.tasky.core.domain.util.EmptyResult
 import com.humberto.tasky.core.domain.util.Result
 import javax.inject.Inject
 
 class EventRepositoryImpl @Inject constructor(
     private val eventDao: EventDao,
-    private val agendaApiService: AgendaApiService
+    private val agendaApiService: AgendaApiService,
+    private val sessionManager: SessionManager
 ) : EventRepository {
 
+    private val localUserId: String?
+        get() = sessionManager.getUserId()
+
     override suspend fun getEvent(eventId: String): Result<AgendaItem, DataError> {
         val eventEntity = eventDao.getEvent(eventId)
-        return eventEntity?.let {
+        return eventEntity?.let { event ->
             val photos = eventDao
                 .getPhotosByKeys(eventEntity.photoKeys)
                 .map { it.toPhoto() }
+            val localAttendee = event.attendees.find { it.userId == localUserId }
+            val eventCreator = event.attendees.find { it.userId == eventEntity.host }
             Result.Success(
                 eventEntity.toEvent(
-                    photos = photos
+                    photos = photos,
+                    eventCreator = eventCreator,
+                    localAttendee = localAttendee
```

Comment:
I see your `EventEntity` has attendees, why then do you pass in the attendees to `toEvent`?
Regardless, if the `eventCreator` is null, then you can consider the data invalid and return an error - the event should always have a creator